rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -717,25 +718,20 @@ def parse_action_file(pkg_name, interface_filename):
         return parse_action_string(pkg_name, action_name, h.read())
 
 
-def parse_action_string(pkg_name, action_name, message_string):
-    lines = message_string.splitlines()
-    separator_indices = [
-        index for index, line in enumerate(lines) if line == SERVICE_REQUEST_RESPONSE_SEPARATOR]
-    if not separator_indices:
-        raise InvalidServiceSpecification(
-            "Could not find separator '%s' between action goal, result and feedback services" %
-            SERVICE_REQUEST_RESPONSE_SEPARATOR)
-    if len(separator_indices) != 3:
+def parse_action_string(pkg_name, action_name, action_string):
+    action_blocks = action_string.split(ACTION_REQUEST_RESPONSE_SEPARATOR)
+    if len(action_blocks) != 2:
```

Comment:
Before it was counting the number of `---` in an action file which should be 2. Now it's counting the sections divided by `---` which is 3.

Reply:
@sloretz I know, it was fixed in `3b7a695`, latest commit.

Reply:
Ah oops, I forgot I clicked on a single commit.